src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -203,7 +199,6 @@ trait ScalaSettings extends AbsScalaSettings
   val Ynogenericsig   = BooleanSetting    ("-Yno-generic-signatures", "Suppress generation of generic signatures for Java.")
   val noimports       = BooleanSetting    ("-Yno-imports", "Compile without importing scala.*, java.lang.*, or Predef.")
   val nopredef        = BooleanSetting    ("-Yno-predef", "Compile without importing Predef.")
-  val noAdaptedArgs   = BooleanSetting    ("-Yno-adapted-args", "Do not adapt an argument list (either by inserting () or creating a tuple) to match the receiver.")
```

Comment:
Kind of sad to see this one go. I only learned about it recently, but I thought it was a good cop.

Reply:
This PR is part provocation to hear any passionate defenses of the defendants :-)

Reply:
I'm quite happy to see that one go. As Scala sharp edges go it's one of the least harmful (you're unlikely to silently miscompile because of it) and most useful (see various bits of arity-polymorphic sugar in shapeless and thereabouts).

Reply:
Here's an easy example of things compiling:

```scala
scala> case class W[A](elem: A)(implicit ordering: Ordering[A])
defined class W

scala> W(1, 2, 3, 4)
res0: W[(Int, Int, Int, Int)] = W((1,2,3,4))
```

And to quote @lrytz in https://github.com/scala/bug/issues/8342:

> Auto-tupling is not mentioned at all in the spec.

Here's another [voice from the wild](https://twitter.com/djspiewak/status/430758303119863808):

<img width="638" alt="AGGGHGHGHGHGHGHGHGHGHGHGHGH!
Scala auto-tupling. Why on earth does this “feature” even exist?" src="https://user-images.githubusercontent.com/184683/38578590-5d5d1fb4-3cd2-11e8-80dc-2dbdd5ca6fad.png">

To which @tpolecat responds "`-Yno-adapted-args`."

I am not so much for or against the removal of `-Yno-adapted-args` per se, but I'd request a more principled solution toward this general problem. Adaptations.scala is already riddled with flags (`settings.future`, `settings.noAdaptedArgs`, `settings.warnAdaptedArgs`).

### option 1

Here's a proposal. Remove zero-arg eta expansion in 2.13 (or under -Xsource:2.14), and also remove auto-tupling under -Xsource:2.14 if Dotty can go along with it.

Reply:
> Here's a proposal. Remove zero-arg eta expansion in 2.13

Yup: https://github.com/scala/scala/pull/6475

> remove auto-tupling
Would be nice...

Reply:
I find that auto-tupling hinders readability, and at least for me it ends up being applied accidentally often enough that I prefer to turn it off. The 22-element limit makes it unusable for arity-abstraction so I don't really buy that argument. So I'm 👎for removing the flag and 👍for removing the feature.

Reply:
Been saying that for 4 years now ;-) https://github.com/scala/bug/issues/3583#issuecomment-292383595

Reply:
One data point: a very nasty issue (https://github.com/scala-js/scala-js/issues/3154, fix in https://github.com/scala-js/scala-js/commit/9428b684197cea946467f9caeaefab616546d356) that could have been avoided if auto-tupling did not exist.

So yup, I'm not a fan of auto-tupling ...

Reply:
It's more restricted in dotty though: http://dotty.epfl.ch/docs/reference/auto-parameter-tupling.html. Looks more acceptable, though I wouldn't mind just spelling it out to avoid the gotcha in other places.

Reply:
In any case, our guiding principle is alignment with Dotty, so, looks like we can at least deprecate auto-tupling in most cases in 2.13 (drop it in 2.14), and implement the same approach as dotty by 2.14.

Reply:
Dotty still has the regular auto tupling which I don't care much about. But the function parameter tupling is something different that doesn't exist in scalac and I find it quite useful.

Reply:
Ah, thanks for the correction. Wasn't aware they still had auto-tupling :-( I would really like to drop that, while I agree the one for function literals seems useful.

Reply:
In an effort for reconciliation, I've opened a thread here https://contributors.scala-lang.org/t/lets-drop-auto-tupling/1799, please comment and subscribe.